openpecha/formatters/otranscribe.py

```diff
@@ -0,0 +1,119 @@
+from openpecha.core.annotations import TranscriptionTimeSpan, Span
+from openpecha.formatters.formatter import BaseFormatter
+from openpecha.core.layer import TranscriptionTimeSpanLayer, LayerEnum
+from openpecha.core.annotations import TranscriptionTimeSpan, Span
+from openpecha.core.pecha import OpenPechaFS
+from openpecha.core.ids import get_initial_pecha_id, get_base_id
+from openpecha.core.metadata import InitialPechaMetadata
+import json
+from bs4 import BeautifulSoup
+
+
+def get_json_from_str(text):
+    otr_json = json.loads(text)
+    return otr_json
+
+
+def parse_html(html):
+    soup = BeautifulSoup(html, "html.parser")
+    return soup
+
+
+def get_all_timestamps(soup):
+    return soup.findAll("span", {"class": "timestamp"})
+
+
+def get_text_between(start, end):
+    text = ""
+    tag = start.next_element.next_element
+    while tag != end:
+        if tag.name is None:
+            text += tag.text
+        tag = tag.next_element
+    return text
+
+
+def get_text_till_end(start):
+    text = ""
+    tag = start.next_element.next_element
+    while tag:
+        if tag.name is None:
+            text += tag.text
+        tag = tag.next_element
+    return text
+
+
+class OTranscribeFormatter(BaseFormatter):
+    """
+    oTranscribe formatter for .otr file.
+    """
+
+    def __init__(
+        self,
+        output_path="./output",
+        metadata=None,
+        media_url=None,
+        time_unit="microsecond",
+    ):
+        super().__init__(output_path, metadata)
+        self.media_url = media_url
+        self.time_unit = time_unit
+
+    def text_preprocess(self, text):
+        otr_json = get_json_from_str(text)
+        return otr_json["text"].strip()
+
+    def build_layers(self, text):
+        transcription_time_span_layer = TranscriptionTimeSpanLayer(
+            media_url=self.media_url,
+            time_unit=self.time_unit,
+        )
+
+        soup = parse_html(text)
+        timestamps = get_all_timestamps(soup)
+        base_text = ""
+        char_walker = 0
+        for start, end in zip(timestamps, timestamps[1:]):
+            time_start = float(start["data-timestamp"]) * 10**6
+            time_end = float(end["data-timestamp"]) * 10**6
+            time_span = Span(start=time_start, end=time_end)
+            text_between = get_text_between(start, end).strip()
+            text_end = char_walker + len(text_between)
+            text_span = Span(start=char_walker, end=text_end)
+            char_walker += len(text_between) + 1
+            transcription_time_span_layer.set_annotation(
+                TranscriptionTimeSpan(span=text_span, time_span=time_span)
+            )
+            base_text += text_between + "\n"
+        # \xa0 is non-breaking space. Replace it with space.
+        base_text = base_text.replace("\xa0", " ")
+        return {
+            LayerEnum.transcription_time_span: transcription_time_span_layer
+        }, base_text
+
+    def create_opf(self, input_path):
```

Comment:
how will you format a input text where all the chapters of text are saved in separate otr. 
-> input-folder
  -chapter01.otr
  -chapter02.otr
You are suppose to have one opf having two bases in this case. how will it be done using current formatter. i am not able to find any loop through which u r accessing those otr

Reply:
I made the changes.